darts/timeseries.py

```diff
@@ -1001,7 +1028,8 @@ def from_series(
         TimeSeries
             A univariate and deterministic TimeSeries constructed from the inputs.
         """
-        df = pd.DataFrame(pd_series)
+        nw_series = nw.from_native(pd_series, series_only=True)
```

Comment:
Similarly as above, if you want to already raise for non series objects, it's possible to do so by flagging `pass_through=False`